masonry/src/render_root.rs

```diff
@@ -570,8 +567,11 @@ impl RenderRoot {
             self.state.focused_widget = new;
             self.root_lifecycle(event);
 
-            // TODO - Handle IME
-            // Send TextFieldFocused(focused_widget) signal
+            self.state.signal_queue.push_back(if new.is_some() {
+                RenderRootSignal::StartIme
+            } else {
+                RenderRootSignal::EndIme
```

Comment:
I think that part is a bit over-simplified.

What happens if you switch between two textboxes? Shouldn't the IME session be restarted, instead of just moved to a different position?

Reply:
No it should not be restarted, however the surround should be updated (which is part of the Android IME work).

Reply:
Though maybe this is more just exposing an issue with conflation of opening the soft keyboard with starting the IME. The issue with restarting it has nothing to do with resetting the IME (which is desirable) but rather relates to opening the soft keyboard. On Wayland, these two things are not separable, but on Android they are separable.
Winit might be able to work around this by coalescing the two `set_ime_allowed` state changes but idk if that introduces more problems.